livekit-rtc/livekit/rtc/video_frame.py

```diff
@@ -29,229 +32,575 @@ def __init__(self, timestamp_us: int,
         self.rotation = rotation
 
 
-class VideoFrameBuffer:
-    def __init__(self, owned_info: proto_video_frame.OwnedVideoFrameBuffer) -> None:
-        self._info = owned_info.info
-        self._ffi_handle = FfiHandle(owned_info.handle.id)
+class VideoFrameBuffer(ABC):
```

Comment:
might be worth unit testing all of the different frame types in this one. Seems like there's a fair amount of math that would be hard to debug if it was wrong.